From db75362a8223344dc8f0746c076cef7e4687b755 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 27 Aug 2018 22:43:22 +0530
Subject: [PATCH 3/3] Fix for Bluetooth device name reset to default name after
 reboot

Reason: When the BT device name is updated is getting saved
to the config data pointer but not to the persistent data
(i.e. bt_config.conf). So, when the reboot is happening it
is not able to get the updated device name from the persistent
data (i.e. bt_config.conf) as during reboot bt_config_flush is
not called. It is only called on the BT Enable Event.

Fix: Saving the BT Device name to persistent data using
btif_config_flush once it is set.

Tracked-On: OAM-67917

Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 btif/src/btif_storage.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/btif/src/btif_storage.cc b/btif/src/btif_storage.cc
index fe00a2b..13dd58d 100644
--- a/btif/src/btif_storage.cc
+++ b/btif/src/btif_storage.cc
@@ -207,10 +207,12 @@ static int prop2cfg(const RawAddress* remote_bd_addr, bt_property_t* prop) {
                             : prop->len;
       strncpy(value, (char*)prop->val, name_length);
       value[name_length] = '\0';
-      if (remote_bd_addr)
+      if (remote_bd_addr) {
         btif_config_set_str(bdstr, BTIF_STORAGE_PATH_REMOTE_NAME, value);
-      else
+      } else {
         btif_config_set_str("Adapter", BTIF_STORAGE_KEY_ADAPTER_NAME, value);
+        btif_config_flush();
+      }
       break;
     }
     case BT_PROPERTY_REMOTE_FRIENDLY_NAME:
-- 
2.17.1

